internal/resource/config/serviceauthentication/service_authentication_resource_gen.go

```diff
@@ -0,0 +1,359 @@
+	client "github.com/pingidentity/pingfederate-go-client/v1210/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+var (
+	_ resource.Resource                = &serviceAuthenticationResource{}
+	_ resource.ResourceWithConfigure   = &serviceAuthenticationResource{}
+	_ resource.ResourceWithImportState = &serviceAuthenticationResource{}
+)
+
+func ServiceAuthenticationResource() resource.Resource {
+	return &serviceAuthenticationResource{}
+}
+
+type serviceAuthenticationResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+func (r *serviceAuthenticationResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_authentication"
+}
+
+func (r *serviceAuthenticationResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+}
+
+type serviceAuthenticationResourceModel struct {
+	AttributeQuery       types.Object `tfsdk:"attribute_query"`
+	ConnectionManagement types.Object `tfsdk:"connection_management"`
+	Jmx                  types.Object `tfsdk:"jmx"`
+	SsoDirectoryService  types.Object `tfsdk:"sso_directory_service"`
+}
+
+func (r *serviceAuthenticationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Resource to manage the service authentication settings.",
+		Attributes: map[string]schema.Attribute{
+			"attribute_query": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"id": schema.StringAttribute{
+						Optional:    true,
+						Description: "Id of the service.",
+					},
+					"shared_secret": schema.StringAttribute{
+						Optional:    true,
+						Sensitive:   true,
+						Description: "Shared secret for the service.",
+					},
+				},
+				Optional:    true,
+				Description: "SAML2.0 attribute query service. Remove the JSON field to deactivate the attribute query service.",
+			},
+			"connection_management": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"id": schema.StringAttribute{
+						Optional:    true,
+						Description: "Id of the service.",
+					},
+					"shared_secret": schema.StringAttribute{
+						Optional:    true,
+						Sensitive:   true,
+						Description: "Shared secret for the service.",
+					},
+				},
+				Optional:    true,
+				Description: "(Deprecated) Connection management service. Remove the JSON field to deactivate the connection management service.",
```

Comment:
A couple of these attributes are described as deprecated in PF's docs. Should we just not include them in the provider from the start?

Reply:
Ideally we should leave them out of the provider.  PF is following a deprecation path to encourage customers and code clients to migrate their functionality and remove those functions, so they can be safely removed from PF later.

However, because we're offering different contexts of the provider based on different product versions, we have additional complexity.  If the feature is deprecated on all of our supported versions, we should remove.  If not, then we may need extra validation (and extra documentation) to warn on the versions where the feature is deprecated, but not warn where the version being used has it not deprecated.

This means we might end up deprecated and removing fields as we un-support older versions of PF, which should also happen on major version releases of the provider